src/avt/Queries/Queries/avtXRayImageQuery.C

```diff
@@ -731,17 +783,79 @@ avtXRayImageQuery::SetOutputRayBounds(const bool &flag)
 //  Method: avtXRayImageQuery::SetFamilyFiles
 //
 //  Purpose:
-//    Set the family files flag.
+//    Set the filename scheme.
 //
 //  Programmer: Eric Brugger
 //  Creation:   May 27, 2015
 //
+//  Modifications:
+//     Justin Privitera, Thu Sep 22 16:46:46 PDT 2022
+//     Instead of setting the familyfiles flag, now it sets the filename scheme
+//     appropriately.
 // ****************************************************************************
 
 void
 avtXRayImageQuery::SetFamilyFiles(const bool &flag)
 {
-    familyFiles = flag;
+    filenameScheme = flag ? FAMILYFILES : NONE;
+}
+
+// ****************************************************************************
+//  Method: avtXRayImageQuery::SetFilenameScheme
+//
+//  Purpose:
+//    Set the filename scheme.
+//
+//  Programmer: Justin Privitera
+//  Creation:   Tue Sep 27 10:52:59 PDT 2022
+// 
+//  Modifications:
+// 
+// ****************************************************************************
+
+void
+avtXRayImageQuery::SetFilenameScheme(int type)
+{
+    if (filenameSchemeValid(filenameScheme))
+        filenameScheme = type;
+    else
+    {
+        std::ostringstream err_oss;
+        err_oss << "Filename scheme " << type << " is invalid.\n";
+        EXCEPTION1(VisItException, err_oss.str());
+    }
+}
+
+// ****************************************************************************
+//  Method: avtXRayImageQuery::SetFilenameScheme
+//
+//  Purpose:
+//    Set the filename scheme.
+//
+//  Programmer: Justin Privitera
+//  Creation:   Tue Sep 27 10:52:59 PDT 2022
+//
+//  Modifications:
+// 
+// ****************************************************************************
+
+void
+avtXRayImageQuery::SetFilenameScheme(const std::string &type)
+{
+    int i = 0;
+    while (i < NUM_FILENAME_SCHEMES)
```

Comment:
If we don't have a match, this loop executes, currently `9999` times.

Reply:
Yeah, it was like that before. Is there a better way to solve this problem?

Reply:
Marking as resolved since I added this to #17791